File: app/src/main/java/com/rst2g1/northkite/ui/smedia/Post.java
package com.rst2g1.northkite.ui.smedia;

public class Post {
    private String username;
    private String profilePicUrl;
    private String postImageUrl;
    private int likes;
    private String description;

    public Post() {
        // Default constructor required for calls to DataSnapshot.getValue(Post.class)
    }

    public Post(String username, String profilePicUrl, String postImageUrl, int likes, String description) {
        this.username = username;
        this.profilePicUrl = profilePicUrl;
        this.postImageUrl = postImageUrl;
        this.likes = likes;
        this.description = description;
    }

    public String getUsername() {
        return username;
    }

    public String getProfilePicUrl() {
        return profilePicUrl;
    }

    public String getPostImageUrl() {
        return postImageUrl;
    }

    public int getLikes() {
        return likes;
    }

    public String getDescription() {
        return description;
    }
}

File: app/src/main/java/com/rst2g1/northkite/ui/smedia/Post.kt
package com.rst2g1.northkite.ui.smedia

data class Post(
    var id: String = "",
    val username: String = "",
    val profilePicUrl: String = "",
    val postImageUrl: String = "",
    var likes: Int = 0,
    val description: String = "",
    val likedBy: MutableList<String> = mutableListOf()
)
